R/model.R

```diff
@@ -369,22 +374,146 @@ compile_method <- function(quiet = TRUE,
   if (is.null(include_paths) && !is.null(private$precompile_include_paths_)) {
     include_paths <- private$precompile_include_paths_
   }
-  if (is.null(dir) && !is.null(private$dir_)) {
-    dir <- absolute_path(private$dir_)
-  } else if (!is.null(dir)) {
-    dir <- absolute_path(dir)
-  }
-  if (!is.null(dir)) {
-    dir <- repair_path(dir)
-    checkmate::assert_directory_exists(dir, access = "rw")
-  }
 
   # temporary deprecation warnings
   if (isTRUE(threads)) {
     warning("'threads' is deprecated. Please use 'cpp_options = list(stan_threads = TRUE)' instead.")
     cpp_options[["stan_threads"]] <- TRUE
   }
 
+  model_name <- sub(" ", "_", paste0(strip_ext(basename(self$stan_file())), "_model"))
+
+  stancflags_val <- NULL
+  if (!is.null(include_paths)) {
+    checkmate::assert_directory_exists(include_paths, access = "r")
+    include_paths <- absolute_path(include_paths)
+    include_paths <- paste0(include_paths, collapse = ",")
+    if (cmdstan_version() >= "2.24") {
+      include_paths_flag <- " --include-paths="
+    } else {
+      include_paths_flag <- " --include_paths="
+    }
+    stancflags_val <- paste0(stancflags_val, include_paths_flag, include_paths, " ")
+  }
+
+  if (!is.null(cpp_options$stan_opencl)) {
+    stanc_options[["use-opencl"]] <- TRUE
+  }
+  if (is.null(stanc_options[["name"]])) {
+    stanc_options[["name"]] <- model_name
+  }
+  if(make_or_stanc=='make'){
+    char_for_stanc_built_options <- "'"
+  }else{
+    char_for_stanc_built_options <- ""
+  }
```

Comment:
Can you add a comment here to remind us in the future why it was necessary to use `char_for_stanc_built_options` differently for make and stanc? Maybe Rok will remember but I will definitely forget ;)

Reply:
So it should really be "\"" for the stanc case, I'll commit that change momentarily. That needs to be a double-quote because `processx::run()` uses single-quotes to encapsulate entries in the list supplied to it's `args` argument. 

The make case is because the entire `STANCFLAGS +=` string gets encapsulated by double quotes at some point, so it's necessary to use a single quote to encapsulate anything inside that string that needs encapsulating. If we can find where the double-quotes are added around the `STANCFLAGS +=` string and change it to single quotes, we can get rid of this conditional.

Reply:
Oh! I figured it out. processx::run() is smart enough to just use whatever doesn't break any existing encapsulation, so it's the one adding the double quotes to the stancflags string. So I'll get rid of this conditional.

Reply:
> So it should really be """ for the stanc case

In that case I think you'll have to use `'"'` with single quotes wrapping the double quote or R won't like it. 


> The make case is because the entire STANCFLAGS += string gets encapsulated by double quotes at some point, so it's necessary to use a single quote to encapsulate anything inside that string that needs encapsulating. If we can find where the double-quotes are added around the STANCFLAGS += string and change it to single quotes, we can get rid of this conditional.

Ok I think I see what you mean. I'm not sure where the double quotes are added at the moment, but we can probably track it down.